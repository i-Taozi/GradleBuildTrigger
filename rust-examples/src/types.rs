///类型
pub fn types() {
    let decimal = 65.4321_f32;
    //不支持隐式转化
    //let integer: u8 = decimal;
    // FIXME ^ Comment out this line

    //显示转化
    let integer = decimal as u8;
    let character = integer as char;

    println!("Casting: {} -> {} -> {}", decimal, integer, character);

    // 1000 already fits in a u16
    println!("1000 as a u16 is: {}", 1000 as u16);

    // 1000 - 256 - 256 - 256 = 232
    // Under the hood, the first 8 least significant bits (LSB) are kept,
    // while the rest towards the most significant bit (MSB) get truncated.
    //println!("1000 as a u8 is : {}", 1000 as u8);
    // -1 + 256 = 255
    println!("  -1 as a u8 is : {}", (-1i8) as u8);
    println!("1000 mod 256 is : {}", 1000 % 256);

    println!(" 128 as a i16 is: {}", 128 as i16);
    //println!(" 128 as a i8 is : {}", 128 as i8);
    //println!("1000 as a u8 is : {}", 1000 as u8);
    //println!(" 232 as a i8 is : {}", 232 as i8);

    literals();
}

fn literals() {
    //类型作为后缀
    let x = 1u8;
    let y = 2u32;
    let z = 3f32;

    //非固定文字，其类型取决于如何使用
    let i = 1;
    let f = 1.0;

    //Returns the size of the pointed-to value in bytes.
    println!("size of `x` in bytes: {}", std::mem::size_of_val(&x));
    println!("size of `y` in bytes: {}", std::mem::size_of_val(&y));
    println!("size of `z` in bytes: {}", std::mem::size_of_val(&z));
    println!("size of `i` in bytes: {}", std::mem::size_of_val(&i));
    println!("size of `f` in bytes: {}", std::mem::size_of_val(&f));
}
