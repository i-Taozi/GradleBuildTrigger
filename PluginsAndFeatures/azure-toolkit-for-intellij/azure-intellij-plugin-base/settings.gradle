rootProject.name = 'azure-intellij-plugin-base'
include ':azure-intellij-plugin-lib'
include ':azure-intellij-plugin-service-explorer'
project(':azure-intellij-plugin-lib').projectDir = new File(settingsDir, IntellijPluginBase + '/azure-intellij-plugin-lib')
project(':azure-intellij-plugin-service-explorer').projectDir = new File(settingsDir, IntellijPluginBase + '/azure-intellij-plugin-service-explorer')

def modules = IntellijPluginModules.split(",")
modules.each { m ->
    include ':' + m
    project(':' + m).projectDir = new File(settingsDir, IntellijPluginBase + '/' + m)
}
