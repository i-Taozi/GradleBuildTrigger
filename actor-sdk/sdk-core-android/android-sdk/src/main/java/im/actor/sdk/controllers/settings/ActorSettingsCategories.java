package im.actor.sdk.controllers.settings;

import java.util.ArrayList;

public class ActorSettingsCategories extends ArrayList<ActorSettingsCategory> {
    public ActorSettingsCategories addCategory(ActorSettingsCategory category) {
        super.add(category);
        return this;
    }
}
