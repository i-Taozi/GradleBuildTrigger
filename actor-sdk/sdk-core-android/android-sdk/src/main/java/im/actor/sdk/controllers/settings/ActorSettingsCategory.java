package im.actor.sdk.controllers.settings;

import android.content.Context;
import android.view.View;

import java.util.ArrayList;

public class ActorSettingsCategory {

    private int iconResourceId = 0;
    private int iconColor = -1;
    private String text;
    private ArrayList<ActorSettingsField> fields = new ArrayList();

    public ActorSettingsCategory(String text) {
        this.text = text;
    }

    public int getIconResourceId() {
        return iconResourceId;
    }

    public ActorSettingsCategory setIconResourceId(int iconResourceId) {
        this.iconResourceId = iconResourceId;
        return this;
    }

    public int getIconColor() {
        return iconColor;
    }

    public ActorSettingsCategory setIconColor(int iconColor) {
        this.iconColor = iconColor;
        return this;
    }

    public ArrayList<ActorSettingsField> getFields() {
        return fields;
    }

    public ActorSettingsCategory setFields(ArrayList<ActorSettingsField> fields) {
        this.fields = fields;
        return this;
    }

    public ActorSettingsCategory addField(ActorSettingsField field) {
        fields.add(field);
        return this;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }
}
