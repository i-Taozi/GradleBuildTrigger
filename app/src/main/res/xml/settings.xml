<?xml version="1.0" encoding="utf-8"?>
<PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
    <RingtonePreference
        android:key="alert_alarm_tone"
        android:showDefault="true"
        android:showSilent="false"
        android:summary="@string/select_the_alarm_tone_used_for_pump_alerts"
        android:title="@string/alarm_tone" />
    <CheckBoxPreference
        android:defaultValue="false"
        android:key="background_sync_enabled"
        android:summary="@string/synchronize_pump_data_automatically_in_the_background"
        android:title="@string/background_sync" />
    <CheckBoxPreference
        android:defaultValue="false"
        android:key="auto_adjust_time"
        android:summary="@string/automatically_adjust_the_pumps_time_if_its_inaccurate"
        android:title="@string/automatically_adjust_pump_time" />

    <PreferenceCategory
        android:title="@string/confirmation">
        <CheckBoxPreference
            android:defaultValue="true"
            android:key="enable_confirmation_challenges"
            android:summary="@string/avoid_inadvertent_bolus_deliveries"
            android:title="@string/confirmation_challenges" />
        <CheckBoxPreference
            android:defaultValue="false"
            android:dependency="enable_confirmation_challenges"
            android:key="confirmation_use_fingerprint"
            android:summary="@string/use_your_fingerprint_to_confirm_actions"
            android:title="@string/use_fingerprint" />
        <CheckBoxPreference
            android:defaultValue="false"
            android:dependency="enable_confirmation_challenges"
            android:key="confirmation_use_pin"
            android:inputType="numberPassword"
            android:summary="@string/use_a_pin_instead_of_a_challenge"
            android:title="@string/use_pin" />
        <Preference
            android:dependency="confirmation_use_pin"
            android:key="confirmation_pin"
            android:summary="@string/change_the_pin_used_for_confirmation"
            android:title="@string/change_pin" />
    </PreferenceCategory>

    <PreferenceCategory
        android:title="@string/connection">
        <Preference
            android:summary="@string/restrict_the_apps_functionality"
            android:title="@string/firewall_options">
            <intent
                android:targetClass="sugar.free.sightremote.activities.FirewallActivity"
                android:targetPackage="sugar.free.sightremote" />
        </Preference>
        <Preference
            android:key="enter_password"
            android:summary="@string/enter_the_password_required_for_remote_controlling_the_pump"
            android:title="@string/enter_password" />
        <Preference
            android:key="delete_pairing"
            android:summary="@string/delete_the_pairing_with_your_pump"
            android:title="@string/delete_pairing" />
    </PreferenceCategory>
</PreferenceScreen>