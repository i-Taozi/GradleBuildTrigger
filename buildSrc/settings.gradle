dependencyResolutionManagement {
    repositories {
        mavenLocal()
        maven {
            name "aliyunmaven"
            url "https://maven.aliyun.com/nexus/content/groups/public/"
        }
        maven {
            name "spring"
            url "https://repo.spring.io/libs-milestone/"
        }
        mavenCentral()
        gradlePluginPortal()
        google()
    }
}
rootProject.name = 'buildSrc'
