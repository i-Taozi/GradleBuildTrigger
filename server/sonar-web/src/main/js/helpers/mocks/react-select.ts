/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { GroupTypeBase, OptionProps } from 'react-select';

export function mockReactSelectOptionProps<
  OptionType,
  IsMulti extends boolean,
  GroupType extends GroupTypeBase<OptionType> = GroupTypeBase<OptionType>
>(
  data: OptionType,
  overrides?: OptionProps<OptionType, IsMulti, GroupType>
): OptionProps<OptionType, IsMulti, GroupType> {
  return {
    ...overrides,
    data
  } as OptionProps<OptionType, IsMulti, GroupType>;
}
