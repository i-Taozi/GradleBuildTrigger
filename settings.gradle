include ':app', ':bankdroid-core', ':bankdroid-legacy', ':bankdroid-interface'
