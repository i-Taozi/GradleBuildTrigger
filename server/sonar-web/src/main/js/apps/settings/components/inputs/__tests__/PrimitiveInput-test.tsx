/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import { mockDefinition, mockSetting } from '../../../../../helpers/mocks/settings';
import { SettingType } from '../../../../../types/settings';
import { DefaultSpecializedInputProps } from '../../../utils';
import PrimitiveInput from '../PrimitiveInput';

it.each(Object.values(SettingType).map(Array.of))(
  'should render correctly for %s',
  (type: SettingType) => {
    const setting = mockSetting({ definition: mockDefinition({ type }) });
    expect(shallowRender({ setting })).toMatchSnapshot();
  }
);

function shallowRender(props: Partial<DefaultSpecializedInputProps> = {}) {
  return shallow(
    <PrimitiveInput
      isDefault={true}
      name="name"
      onChange={jest.fn()}
      setting={mockSetting()}
      value={['foo']}
      {...props}
    />
  );
}
