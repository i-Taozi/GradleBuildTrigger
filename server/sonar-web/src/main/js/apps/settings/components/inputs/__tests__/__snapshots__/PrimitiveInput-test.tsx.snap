// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly for BOOLEAN 1`] = `
<InputForBoolean
  isDefault={true}
  name="name"
  onChange={[MockFunction]}
  setting={
    Object {
      "definition": Object {
        "category": "foo category",
        "fields": Array [],
        "key": "foo",
        "options": Array [],
        "subCategory": "foo subCat",
        "type": "BOOLEAN",
      },
      "hasValue": true,
      "inherited": true,
      "key": "foo",
      "value": "42",
    }
  }
  value={
    Array [
      "foo",
    ]
  }
/>
`;

exports[`should render correctly for FLOAT 1`] = `
<InputForNumber
  isDefault={true}
  name="name"
  onChange={[MockFunction]}
  setting={
    Object {
      "definition": Object {
        "category": "foo category",
        "fields": Array [],
        "key": "foo",
        "options": Array [],
        "subCategory": "foo subCat",
        "type": "FLOAT",
      },
      "hasValue": true,
      "inherited": true,
      "key": "foo",
      "value": "42",
    }
  }
  value={
    Array [
      "foo",
    ]
  }
/>
`;

exports[`should render correctly for INTEGER 1`] = `
<InputForNumber
  isDefault={true}
  name="name"
  onChange={[MockFunction]}
  setting={
    Object {
      "definition": Object {
        "category": "foo category",
        "fields": Array [],
        "key": "foo",
        "options": Array [],
        "subCategory": "foo subCat",
        "type": "INTEGER",
      },
      "hasValue": true,
      "inherited": true,
      "key": "foo",
      "value": "42",
    }
  }
  value={
    Array [
      "foo",
    ]
  }
/>
`;

exports[`should render correctly for JSON 1`] = `
<InputForJSON
  isDefault={true}
  name="name"
  onChange={[MockFunction]}
  setting={
    Object {
      "definition": Object {
        "category": "foo category",
        "fields": Array [],
        "key": "foo",
        "options": Array [],
        "subCategory": "foo subCat",
        "type": "JSON",
      },
      "hasValue": true,
      "inherited": true,
      "key": "foo",
      "value": "42",
    }
  }
  value={
    Array [
      "foo",
    ]
  }
/>
`;

exports[`should render correctly for LICENSE 1`] = `
<InputForString
  isDefault={true}
  name="name"
  onChange={[MockFunction]}
  setting={
    Object {
      "definition": Object {
        "category": "foo category",
        "fields": Array [],
        "key": "foo",
        "options": Array [],
        "subCategory": "foo subCat",
        "type": "LICENSE",
      },
      "hasValue": true,
      "inherited": true,
      "key": "foo",
      "value": "42",
    }
  }
  value={
    Array [
      "foo",
    ]
  }
/>
`;

exports[`should render correctly for LONG 1`] = `
<InputForNumber
  isDefault={true}
  name="name"
  onChange={[MockFunction]}
  setting={
    Object {
      "definition": Object {
        "category": "foo category",
        "fields": Array [],
        "key": "foo",
        "options": Array [],
        "subCategory": "foo subCat",
        "type": "LONG",
      },
      "hasValue": true,
      "inherited": true,
      "key": "foo",
      "value": "42",
    }
  }
  value={
    Array [
      "foo",
    ]
  }
/>
`;

exports[`should render correctly for PASSWORD 1`] = `
<InputForPassword
  isDefault={true}
  name="name"
  onChange={[MockFunction]}
  setting={
    Object {
      "definition": Object {
        "category": "foo category",
        "fields": Array [],
        "key": "foo",
        "options": Array [],
        "subCategory": "foo subCat",
        "type": "PASSWORD",
      },
      "hasValue": true,
      "inherited": true,
      "key": "foo",
      "value": "42",
    }
  }
  value={
    Array [
      "foo",
    ]
  }
/>
`;

exports[`should render correctly for PROPERTY_SET 1`] = `
<InputForString
  isDefault={true}
  name="name"
  onChange={[MockFunction]}
  setting={
    Object {
      "definition": Object {
        "category": "foo category",
        "fields": Array [],
        "key": "foo",
        "options": Array [],
        "subCategory": "foo subCat",
        "type": "PROPERTY_SET",
      },
      "hasValue": true,
      "inherited": true,
      "key": "foo",
      "value": "42",
    }
  }
  value={
    Array [
      "foo",
    ]
  }
/>
`;

exports[`should render correctly for SINGLE_SELECT_LIST 1`] = `
<Wrapped
  isDefault={true}
  name="name"
  onChange={[MockFunction]}
  setting={
    Object {
      "definition": Object {
        "category": "foo category",
        "fields": Array [],
        "key": "foo",
        "options": Array [],
        "subCategory": "foo subCat",
        "type": "SINGLE_SELECT_LIST",
      },
      "hasValue": true,
      "inherited": true,
      "key": "foo",
      "value": "42",
    }
  }
  value={
    Array [
      "foo",
    ]
  }
/>
`;

exports[`should render correctly for STRING 1`] = `
<InputForString
  isDefault={true}
  name="name"
  onChange={[MockFunction]}
  setting={
    Object {
      "definition": Object {
        "category": "foo category",
        "fields": Array [],
        "key": "foo",
        "options": Array [],
        "subCategory": "foo subCat",
        "type": "STRING",
      },
      "hasValue": true,
      "inherited": true,
      "key": "foo",
      "value": "42",
    }
  }
  value={
    Array [
      "foo",
    ]
  }
/>
`;

exports[`should render correctly for TEXT 1`] = `
<InputForText
  isDefault={true}
  name="name"
  onChange={[MockFunction]}
  setting={
    Object {
      "definition": Object {
        "category": "foo category",
        "fields": Array [],
        "key": "foo",
        "options": Array [],
        "subCategory": "foo subCat",
        "type": "TEXT",
      },
      "hasValue": true,
      "inherited": true,
      "key": "foo",
      "value": "42",
    }
  }
  value={
    Array [
      "foo",
    ]
  }
/>
`;
